process/modules/converb_to_conj.py

```diff
@@ -0,0 +1,180 @@
+from process.module import ParaphraseModule
+from process.preprocessing_utils import PreprocessingUtils
+from stanza.utils.conll import CoNLL
+
+import stanza
+import pymorphy2 
+morph = pymorphy2.MorphAnalyzer()
+import pyconll
+import udon2
+
+class ConverbToConjuctionModule(ParaphraseModule):
+    def __init__(self, name="capitalize_nouns") -> None:
+        super().__init__(name=name)
+    
+    def load(self, preproc_utils: PreprocessingUtils) -> None:
+        # load any tools as `preproc_utils` attributes
+        preproc_utils.stanza_model = stanza.Pipeline('ru', processors='tokenize,pos,lemma,depparse')
+        self.loaded = True
+
+    def converb_parser(self, sentence, preproc_utils: PreprocessingUtils):
+        data = {}
+        parsed_data = preproc_utils.stanza_model(sentence)
+        output_file_name = "participle.conllu"
+        CoNLL.write_doc2conll(parsed_data, output_file_name)
+        participle_data = pyconll.load_from_file("participle.conllu")
+        roots = udon2.Importer.from_stanza(parsed_data.to_dict()) 
+        root = roots[0]  
+        selection = root.select_having("feats", "VerbForm=Conv") # выбираем все узлы, содержащие деепричастие
+        for node in selection:
+            subtree = node.get_subtree_text().replace(node.form, '', 1).strip(',.')
+            data['converb'] = node.form
+            inf = False
+            for feat in node.parent.feats:
+                if feat[0] == 'VerbForm' and feat[1] == 'Inf':
+                    inf = True
+            if inf:
+                data['head'] = node.parent.parent.form
+            else:
+                data['head'] = node.parent.form
+            data['dep'] = subtree
+            head_sub = node.parent.get_subtree_text().split()
+            data['agent'] = ''
+            ag_word = ''
+            if len(node.parent.get_by('deprel', 'nsubj')) != 0:
+                agent = node.parent.get_by('deprel', 'nsubj')[0]
+                ag_word = agent.form
+                data['agent'] = agent.get_subtree_text()
+            elif len(node.parent.parent.get_by('deprel', 'nsubj')) != 0:
+                agent = node.parent.parent.get_by('deprel', 'nsubj')[0]
+                ag_word = agent.form
+                data['agent'] = agent.get_subtree_text()
+            data['ag_word'] = ag_word
+        return data
+
+    def chose_converb(self, token):
```

Comment:
```suggestion
    def chose_converb(self, token, morph):
```

`morph` нужно будет передать как аргумент, если убрать его локально

Reply:
Для функции `normal_form` это все тоже актуально (на всякий случай пишу)